README.md
# Profiling of Peripheral Blood Monocytes using CyTOF

CyTOF enables the profiling of up to 100 parameters at the single cell level. Bodenmiller *et al* published the profiling of **P**eripheral **B**lood **M**ononuclear **C**ells (PBMCs) using 9 phenotypic and 14 functional markers. PBMCS are stimulated with 12 treatments and treated with 8 doses of various inhibitors, then profiled using CyTOF. PBMCs are manually gated using the phenotypic markers, identifying 14 populations for which the activity of different signaling networks can be assessed using the functional markers. More details are available in [Bodenmiller *et al* Nat Biotech 2012](https://www.nature.com/articles/nbt.2317/).

To assess the efficiency of high-dimensional data analysis and visualization algorithms, the samples corresponding to untreated PBMCs, either unstimulated or after stimulation with BCR/FcR-XL, PMA/Ionomycin
or vanadate are made available as 2 datasets:

 * `refMat` the signal intensity for each of the 23 channels, for 15792 untreated cells
 * `untreatedMat` the signal intensity for each of the 23 channels, for 36144 cells after stimulation with BCR/FcR-XL, PMA/Ionomycin
or vanadate

For each dataset there are 3 additional matrix:

 * `refPhenotMat` and `untreatedPhenoMat` the signal intensity for the 9 phenotypic channels
 * `refFuncMat` and `untreatedFuncMat` the signal intensity for the 14 functional channels
 * `refAnnots` and `untreatedAnnots` the source file, cell type assignment and stimulation (where appropriate)

Install it from github with:

```{r}
devtools::install_github("yannabraham/bodenmiller")
```

## Travis Build Status

[![Travis-CI Build Status](https://travis-ci.org/yannabraham/bodenmiller.svg?branch=master)](https://travis-ci.org/yannabraham/bodenmiller)


NEWS.md
# 2021-03-28

Use a more efficient compression.


R/reference.R
#' Reference Matrix (Phenotypic)
#'
#' a \code{matrix} containing the signal intensity for phenotypic markers measured using CyTOF
#' on different cell types. See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a matrix with 15792 rows and 9 variables:
#' \itemize{
#'    \item{ CD20 }
#'    \item{ IgM }
#'    \item{ CD4 }
#'    \item{ CD33 }
#'    \item{ HLA-DR }
#'    \item{ CD14 }
#'    \item{ CD7 }
#'    \item{ CD3 }
#'    \item{ CD123 }
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"refPhenoMat"

#' Reference Matrix (Functional)
#'
#' a \code{matrix} containing the signal intensity for functional markers measured using CyTOF
#' on different cell types. See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a matrix with 15792 rows and 14 variables:
#' \itemize{
#'    \item{ pStat1 }
#'    \item{ pSlp76 }
#'    \item{ pBtk }
#'    \item{ pPlcg2 }
#'    \item{ pErk }
#'    \item{ pLat }
#'    \item{ pS6 }
#'    \item{ pNFkB }
#'    \item{ pp38 }
#'    \item{ pStat5 }
#'    \item{ pAkt }
#'    \item{ pSHP2 }
#'    \item{ pZap70 }
#'    \item{ pStat3 }
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"refFuncMat"

#' Reference Annotation
#'
#' A \code{data.frame} containing the source file and cell type for every cell in
#' \code{\link{refPhenoMat}} and \code{\link{refFuncMat}}. See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a data.frame with 15792 rows and 2 variables:
#' \itemize{
#'    \item{Source}{the name of the source (fcs) file}
#'    \item{Cells}{the cell type for the corresponding row in \code{\link{refPhenoMat}}
#'      or \code{\link{refFuncMat}}}
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"refAnnots"


data-raw/process_reference.R
## load the required libraries
library(flowCore)
library(stringr)

## define markers
pheno.markers <- c('CD20','IgM','CD4','CD33','HLA-DR','CD14','CD7','CD3','CD123')
func.markers <- c('pStat1','pSlp76','pBtk','pPlcg2','pErk','pLat','pS6','pNFkB','pp38','pStat5','pAkt','pSHP2','pZap70','pStat3')

## List & load FCS files into a flowSet
reference.files <- dir('data-raw',pattern='H05.fcs')

refSet <- read.flowSet(reference.files,path="data-raw")
pData(refSet)$Cells <- sapply(pData(refSet)$name,function(x) str_split(x,'_')[[1]][2])

## create a cell level annotation
refAnnots <- data.frame(
  Source = unlist(lapply(sampleNames(refSet),
                         function(x) rep(x,nrow(refSet[[x]])))),
  Cells = unlist(lapply(sampleNames(refSet),
                        function(x) rep(pData(refSet)[x,'Cells'],nrow(refSet[[x]]))))
)

## Extract channel data
refMat <- fsApply(refSet,function(ff) {
    cur.mat <- exprs(ff)
    # fix the dimnames
    dimnames(cur.mat)[[2]] <- make.names(pData(parameters(ff))$desc)
    cur.mat <- cur.mat[,make.names(c(pheno.markers,func.markers))]
    return(cur.mat)
  }
)

# transform using arcsinh
refMat <- asinh(refMat/5)

refPhenoMat <- refMat[,make.names(pheno.markers)]
refFuncMat <- refMat[,make.names(func.markers)]

devtools::use_data(refAnnots, refPhenoMat, refFuncMat,overwrite=TRUE)


data-raw/process_untreated.R
## load the required libraries
library(flowCore)
library(stringr)

## define markers
pheno.markers <- c('CD20','IgM','CD4','CD33','HLA-DR','CD14','CD7','CD3','CD123')
func.markers <- c('pStat1','pSlp76','pBtk','pPlcg2','pErk','pLat','pS6','pNFkB','pp38','pStat5','pAkt','pSHP2','pZap70','pStat3')

## List & load FCS files into a flowSet
untreated.files <- dir('data-raw',pattern='fcs')

# remove the reference files
untreated.files <- untreated.files[!str_detect(untreated.files,'H05')]

untreatedSet <- read.flowSet(untreated.files,path="data-raw")
pData(untreatedSet)$Cells <- sapply(pData(untreatedSet)$name,function(x) str_split(x,'_')[[1]][2])

## create a cell level annotation
untreatedAnnots <- data.frame(
  Source = unlist(lapply(sampleNames(untreatedSet),
                         function(x) rep(x,nrow(untreatedSet[[x]])))),
  Treatment = unlist(lapply(sampleNames(untreatedSet),
                     function(x) {
                        well <- str_split(str_split(x,'\\.')[[1]][1],'_')[[1]][3]
                        if(well=='H01') {
                          return(rep('vanadate',nrow(untreatedSet[[x]])))
                        } else if(well=='H08') {
                          return(rep('BCR/FcR-XL',nrow(untreatedSet[[x]])))
                        } else if(well=='H12') {
                          return(rep('PMA/Ionomycin',nrow(untreatedSet[[x]])))
                        } else {
                          return(rep('NA',nrow(untreatedSet[[x]])))
                        }
                      }
                     )),
  Cells = unlist(lapply(sampleNames(untreatedSet),
                        function(x) rep(pData(untreatedSet)[x,'Cells'],nrow(untreatedSet[[x]]))))
)

## Extract channel data
untreatedMat <- fsApply(untreatedSet,function(ff) {
    cur.mat <- exprs(ff)
    # fix the dimnames
    dimnames(cur.mat)[[2]] <- make.names(pData(parameters(ff))$desc)
    cur.mat <- cur.mat[,make.names(c(pheno.markers,func.markers))]
    return(cur.mat)
  }
)

# transform using arcsinh
untreatedMat <- asinh(untreatedMat/5)

untreatedPhenoMat <- untreatedMat[,make.names(pheno.markers)]
untreatedFuncMat <- untreatedMat[,make.names(func.markers)]

devtools::use_data(untreatedAnnots, untreatedPhenoMat, untreatedFuncMat,overwrite=TRUE)


R/untreated.R
#' Untreated Samples with 4 Different Treatments (Phenotypic)
#'
#' a \code{matrix} containing the signal intensity for phenotypic markers measured using CyTOF
#' on different cell types, after stimulation with BCR/FcR-XL, PMA/Ionomycin
#' and vanadate. See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a matrix with 15792 rows and 9 variables:
#' \itemize{
#'    \item{ CD20 }
#'    \item{ IgM }
#'    \item{ CD4 }
#'    \item{ CD33 }
#'    \item{ HLA-DR }
#'    \item{ CD14 }
#'    \item{ CD7 }
#'    \item{ CD3 }
#'    \item{ CD123 }
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"untreatedPhenoMat"

#' Untreated Samples with 4 Different Treatments (Functional)
#'
#' a \code{matrix} containing the signal intensity for functional markers measured using CyTOF
#' on different cell types, after stimulation with BCR/FcR-XL, PMA/Ionomycin
#' and vanadate. See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a matrix with 15792 rows and 14 variables:
#' \itemize{
#'    \item{ pStat1 }
#'    \item{ pSlp76 }
#'    \item{ pBtk }
#'    \item{ pPlcg2 }
#'    \item{ pErk }
#'    \item{ pLat }
#'    \item{ pS6 }
#'    \item{ pNFkB }
#'    \item{ pp38 }
#'    \item{ pStat5 }
#'    \item{ pAkt }
#'    \item{ pSHP2 }
#'    \item{ pZap70 }
#'    \item{ pStat3 }
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"untreatedFuncMat"

#' Untreated Samples with 4 Different Treatments (Annotation)
#'
#' A \code{data.frame} containing the source file, cell type and stimulation for every cell in
#' \code{\link{refPhenoMat}} and \code{\link{refFuncMat}}.
#' Stimulations include BCR/FcR-XL, PMA/Ionomycin and vanadate.See
#' \href{https://www.nature.com/articles/nbt.2317/}{Bodenmiller et al 2012}
#' for details.
#'
#' @format a data.frame with 15792 rows and 2 variables:
#' \itemize{
#'    \item{Source}{the name of the source (fcs) file}
#'    \item{Treatment}{the treatment for the corresponding row in \code{\link{refPhenoMat}}
#'      or \code{\link{refFuncMat}}}
#'    \item{Cells}{the cell type for the corresponding row in \code{\link{refPhenoMat}}
#'      or \code{\link{refFuncMat}}}
#' }
#' @source \url{https://reports.cytobank.org/105/v2/}
"untreatedAnnots"
